### ISSUE TITLE
App does not ask for storage permission

### ISSUE THREAD

**Issue by mxhc** [GitJournal/GitJournal#925]
OS: Android 13 MIUI Global 14.0.3

App does not ask for storage permission so I can't grant it. Therefore I can't select 'Store repo externaly', it fails with 'External Permission Failed' message.
When I go to the android settings > apps > gitjournal > app info > app permissions, there is no permissions allowed or not allowed for this app.

**Comment by andrewmcnaughton** [comment]
Exact same issue.

**Comment by htxryan** [comment]
Same here.

**Comment by thejohneason** [comment]
Same issue here, GitJournal version 1.8.0+3741

Android 14 on a Samsung Galaxy S21 Ultra

**Comment by DleanJeans** [comment]
+1 Android 13 on Samsung S20 FE

**Comment by JakubKoralewski** [comment]
https://github.com/GitJournal/GitJournal/blob/7b83f3a43e8374b53ab445bed9d39e81c020be41/lib/settings/settings_storage.dart#L227
https://pub.dev/packages/permission_handler#requesting-storage-permissions-always-returns-denied-on-android-13-what-can-i-do

> Request the MANAGE_EXTERNAL_STORAGE permission only when your app can't effectively make use of the more privacy-friendly APIs, such as the [Storage Access Framework](https://developer.android.com/training/data-storage/shared/documents-files) or the [Media Store API](https://developer.android.com/training/data-storage/shared/media). Your app's usage of the permission must fall within permitted uses and must be directly tied to the core functionality of the app. If your app includes a use case similar to any of the following, it's likely that it can request the MANAGE_EXTERNAL_STORAGE permission:
>
>    File managers
>    Backup and restore apps
>    Anti-virus apps
>    Document management apps
>    On-device file search
>    Disk and file encryption
>    [Device-to-device data migration](https://developer.android.com/guide/topics/data/testingbackup#TestingTransfer)

I think it would fall under both document management and device-to-device data migration so passing through permitted uses should be fine? @vHanda

**Comment by SheepCommander** [comment]
same issue..

**Comment by jonatan-ivanov** [comment]
It seems this is the issue by far voted the highest (👍🏼), also the followings are connected:
- https://github.com/GitJournal/GitJournal/issues/964
- https://github.com/GitJournal/GitJournal/issues/925
- https://github.com/GitJournal/GitJournal/issues/886
- https://github.com/GitJournal/GitJournal/issues/879
- https://github.com/GitJournal/GitJournal/issues/646

Is there a way to fix this?

**Comment by StegSchreck** [comment]
Same issue on Google Pixel 7 with Android 14

**Comment by h4vocdev** [comment]
Same issue, also I was curious what this function really does? 

I am looking for away to access the file directly on the phone with another app, is that even possible?

**Comment by jonatan-ivanov** [comment]
@h4vocdev Yes, it is possible on older Android versions, that's how I used to use it. Unfortunately, the app does not work with newer versions of Android. :(

**Comment by jonatan-ivanov** [comment]
@vHanda This is by far the highest voted issue (plus a few other connected issues in https://github.com/GitJournal/GitJournal/issues/925#issuecomment-2132826316). Is there a chance you can take a look or you can review and merge a PR if someone fixes it?

**Comment by kamzil** [comment]
+1

**Comment by StalwartS0ul** [comment]
It just keeps saying "External Storage Permission Failed" whenever I tap Store Repo Externally.

**Comment by tasmo** [comment]
Same on Android 15.

**Comment by e-lu6a** [comment]
also having this issue!

**Comment by marnanel** [comment]
Same here!

**Comment by CalebSlack** [comment]
Same here

**Comment by EthraZa** [comment]
Android 14 (Motorola Edge 30) same problem

**Comment by jonatan-ivanov** [comment]
@vHanda ping? :)
